### PR TITLE
Docs section update.

### PR DESCRIPTION
**Pull request by TomPohys**
Updated `Docs` section

- remove old TM2 articles
- update all categories (generate, host, style, website, raster)
- switch categories
- use Generate own map as `Docs` root
- a reference to a `mobile-app` page
- update text
- format Markdown

### CODE REVIEW

File: docs/generate/create-custom-extract.md [hunk 1]
```diff
@@ -3,19 +3,18 @@ layout: docs
 category: generate
 order: 4
 title: Create custom map extract
-description: Create custom map extract of an area of your choice
+description: Create a custom map extract of an area of your choice
 ---
 
-If you need an extract which is not included on the [downloads page](http://openmaptiles.org/downloads), you need to download the planet file and create your own extract.
+If you need an extract that is not included on the [downloads page](http://openmaptiles.org/downloads), you need to download the planet file and create your own MBTiles extract. If you need a smaller area, then it is available as OSM.PBF extract, e.g., on [Geofabrik](https://geofabrik.de/). Extracting could be done before importing into a database.
```
[the remark this answers is not in the document]
Comment (TomPohys) [reply docs/generate/create-custom-extract.md:9]: Make sense, thanks!

**Review by xkomczax — APPROVED** [review]

File: docs/host/maptiler-cloud.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+---
+layout: docs
+category: host
+title: Ready to use API
+description: Serve raster and vector tiles with MapTiler Cloud
+order: 1
+---
+## MapTiler Cloud
+
+We are providing a **free reliable maps API** with maps based on **OpenMapTiles schema** with **OpenStreetMap data**. The whole planet is **updated weekly** with **vector** and **raster** tiles available for everyone.
```
Comment (jachym) [inline docs/host/maptiler-cloud.md:10]: Consider 

MapTiler a.g. is providing


Maybe adding logo somewher?

File: docs/host/maptiler-cloud.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+---
+layout: docs
+category: host
+title: Ready to use API
```
Comment (jachym) [inline docs/host/maptiler-cloud.md:4]: Not sure with the title 


is this for SEO?

**Review by jachym — COMMENTED** [review]
small suggestions

File: docs/host/maptiler-cloud.md (hunk 2 above)
Comment (TomPohys) [reply docs/host/maptiler-cloud.md:10]: Thanks - fix in https://github.com/openmaptiles/www.openmaptiles.org/commit/bee7ff1f41ac243219cc4275990ea17d332cc418.

**Pull request merged**